source/MaterialXRuntime/RtCollection.cpp

```diff
@@ -27,6 +28,7 @@ RtPrim RtCollection::createPrim(const RtToken& typeName, const RtToken& name, Rt
     {
         throw ExceptionRuntimeError("Type names mismatch when creating prim '" + name.str() + "'");
     }
+    PvtPath::throwIfNotRoot(parent.getPath(), typeName.str() + " cannot be created inside of a compound");
```

Comment:
I would remove the word "compound" as it is a VNN name which has no meaning the runtime. Instead maybe use something like it the type can only be created at the top / root level. Then you get rid of the double negative in the logic. (It's bad if it's not...).